src/HookBidPool.sol

```diff
@@ -227,7 +231,7 @@ contract HookBidPool is EIP712, ReentrancyGuard, AccessControl {
         feeBips = _feeBips;
         feeRecipient = _feeRecipient;
         protocol = IHookProtocol(_protocol);
-        setAddressForEipDomain(_protocol);
+        setAddressForEipDomain(address(this));
```

Comment:
M-04

Reply:
Since this is only set in the constructor and not again, you can make the storage value EIP712_DOMAIN_SEPARATOR in EIP712.sol immutable. It'll get saved in bytecode at construction and save gas each time its used vs reading from a storage slot.

Reply:
To solve this, we just migrated to the OZ eip712 library as this library is used elsewhere within the protocol (beyond the scope of this audit).

29d954464a89dc5e65a9494c920474d6a79cdeb7 is the commit hash of just this change

Reply:
Confirmed, this looks perfect.